### ISSUE TITLE
Not working with HTTPS - invalid certificate

### ISSUE THREAD

**Issue by iptvcld** [ghostersk/catchall-webbrowsing#1]
Would you know how i can get this installed on Debian? Same server where AdGuardHome is on?

**Comment by ghostersk** [comment]
Hi, this is supposed to run in Docker container. https://docs.docker.com/engine/install/debian/
And as far as I know, you are not able to choose what port Adguard will send it, it sends always to IP only. ( usually 80 or 443 or port what was the original query)
So you can either run it in docker, or other option would be to create virtual interface on debian, where nothing will be bound to port 80 and 443, they you can skip running the docker container, and just execute the python code as is the startup.sh scrip but bind it to that virtual IP on debian.
 https://serverfault.com/questions/66901/adding-a-second-ip-to-debian-server
 You just need to make sure, if you have anything on the debian server using port 80 and 443, it is not using this virtual ip...
 Because by default programs run on all interfaces  0.0.0.0 ....

**Comment by iptvcld** [comment]
Can i install the docker via unraid?

**Comment by ghostersk** [comment]
yes.
I never used uraid, it may be easier get on it portainer and do the other docker containers from there possibly.
But it is possible to use docker on uraid
https://www.youtube.com/watch?v=lu9o-y2ZGNM

**Comment by iptvcld** [comment]
I added it into unraid docker but I am getting this error
[+] Pulling 1/1
 ✔ catchall Pulled                                                                                                                                            0.3s 
network backend declared as external, but could not be found

**Comment by iptvcld** [comment]
![image](https://github.com/ghostersk/catchall-webbrowsing/assets/34818686/4c9db90c-64ca-425d-9b5c-51518e7e3ad0)

**Comment by iptvcld** [comment]
[+] Pulling 9/9
 ✔ catchall 8 layers [⣿⣿⣿⣿⣿⣿⣿⣿]      0B/0B      Pulled                                                                                                       10.6s 
   ✔ c6cf28de8a06 Pull complete                                                                                                                               0.5s 
   ✔ 891494355808 Pull complete                                                                                                                               0.4s 
   ✔ 6582c62583ef Pull complete                                                                                                                               0.5s 
   ✔ bf2c3e352f3d Pull complete                                                                                                                               1.5s 
   ✔ a99509a32390 Pull complete                                                                                                                               0.7s 
   ✔ d46a03def8d9 Pull complete                                                                                                                               0.8s 
   ✔ 4429b810e09e Pull complete                                                                                                                               0.9s 
   ✔ 2a4ca5af09fa Pull complete                                                                                                                               1.1s 
[+] Running 2/2
 ✔ Network catchall_bridge  Created                                                                                                                           0.0s 
 ✔ Container catchall       Created                                                                                                                           0.1s 
Error response from daemon: driver failed programming external connectivity on endpoint catchall (fa721eebfdcd1eb97f1eceb1700d16a62dd7ffddbdc78577f09ad695b31a19e1): Error starting userland proxy: listen tcp4 0.0.0.0:443: bind: address already in use

**Comment by ghostersk** [comment]
backend is my custom network, you must create one or use the bridge.
```
docker network create \
  --driver=bridge \
  --subnet=10.10.10.0/24 \
  --gateway=10.10.10.1 \
  --attachable=true \
  --opt com.docker.network.bridge.name=backend \
  backend

```

And the second issue is that you already have something on port 443 where you are deploying this.
as I said either bind it to specific IP address or make sure nothing is listening on ports 443 and 80 where you are deploying it.
to see what runs on what ports you can use `netstat` i think it is either `netstat -ano` or `netstat -tulnp`

**Comment by iptvcld** [comment]
@ghostersk I ended up running the .sh it self on Debian but this is what i see on the VM side
![image](https://github.com/ghostersk/catchall-webbrowsing/assets/34818686/4d648313-8288-4bec-a068-17fe0d365932)

and when i try access the site via 443 or 80, I just see site cannot be reached

**Comment by ghostersk** [comment]
I think you did not setup the URL where it will be redirected.
As the default behaviour is to go to the IP address of the container and then redirect to URL what has valid SSL certificate, where all the records will be shown.
![Screenshot_20240529_062241](https://github.com/ghostersk/catchall-webbrowsing/assets/68815071/1df58ff5-b662-4419-84a0-640196ddb6b9)
![Screenshot_20240529_062321](https://github.com/ghostersk/catchall-webbrowsing/assets/68815071/41d745a4-c834-44cf-85df-aff67f0b2778)

the value of webblocker_host in app.py needs to be real URL what is accessible from where you are testing it.
I used for it NGINX Proxy manager and made self signed certificates what I added to my PC.
302 code in your example is redirect, so the first part seems to work.

**Comment by iptvcld** [comment]
@ghostersk thank you will readjust this in the morning and if I may ask how resources intensive is your app for it to be running all the time? Does it catch all the blocks that come its way and get overwhelmed ?

**Comment by ghostersk** [comment]
this is not blocking anything.
This only gets redirected any page what Adguard would block, but instead of giving ip of nothing, it will be send to this app so overtime you will see what was blocked and when it gets blocked it also shows in web browser.
Usually you just get generic website that it could not connect - that can be for many reasons.
This should be very minimal app...

**Comment by iptvcld** [comment]
@ghostersk also is there something on the lighter side I can do in order to perform the same task? I tried doing a web server and attempted to forward anything from the IP to https but couldn’t not get that work for every site. Some site it would work and some sites when I block for example bing.com it would not redirect. I am option to suggestions, been trying to make this work for a week now

**Comment by ghostersk** [comment]
if I am not mistaken you have same issue like here:
![Screenshot_20240529_173807](https://github.com/ghostersk/catchall-webbrowsing/assets/68815071/96d6029e-cbdd-4342-bf31-e397aa3fe629)
This is because when Adguard gives the IP address of the container where Catch all is running, the domain what is blocked will be using more then likely HTTPS, and as my/your ssl certificate is not valid for all the domains it fails as per the screenshot.
There are only few things what could fix it, however i do not have time to do it atm.
1. you would need to have some wildcard certificate for everything - not sure if this is possible
2. ignore bad certificates in web browser - look this up how, for your browser  ( not recommended for security reasons)
3. make the app to change https to http, then it will not block it, as it does not use certificate.

If you do not have error as above, I would need more details what is wrong as I am not sure what you describing.
This is what I was able to test

**Comment by iptvcld** [comment]
This is correct, this is what I see for some sites and for other sites the redirect works fine! I also have NextDNS which i am testing and with their install cert, it blocks this site and shows the custom block page no issues.  I am trying to get the same feel from NextDNS block page but make it work for AGH as i like it better

**Comment by iptvcld** [comment]
You think something like can be caught and fixed for sites like that?
